src/java.base/share/classes/jdk/internal/util/StaticProperty.java

```diff
@@ -66,6 +69,11 @@ private StaticProperty() {}
         USER_DIR  = getProperty(props, "user.dir");
         USER_NAME = getProperty(props, "user.name");
         JAVA_IO_TMPDIR = getProperty(props, "java.io.tmpdir");
+
+        if (!(new File(JAVA_IO_TMPDIR)).isDirectory()) {
+            System.err.println("WARNING: java.io.tmpdir location does not exist");
+        }
+
```

Comment:
System.out/err relies on encoding properties are initialized, so I don't think this would work.